v2/pkg/subscraping/sources/hackertarget/hackertarget_test.go

```diff
@@ -0,0 +1,67 @@
+package hackertarget_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping/sources/hackertarget"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHackerTargetWithActualDomain(t *testing.T) {
+	//given
+	ctx := context.Background()
+	domain := "projectdiscovery.io"
+	keys := &subscraping.Keys{}
+	session, err := subscraping.NewSession(domain, keys, "", 0, 10)
```

Comment:
the signature of the `NewSession` has changed. The keys are now stored in the providers themselves, making them self-contained.